test/metabase/models/card_test.clj

```diff
@@ -377,3 +377,23 @@
       (is (empty? (serdes.base/serdes-descendants "Card" (:id card1))))
       (is (= #{["Card" (:id card1)]}
              (serdes.base/serdes-descendants "Card" (:id card2)))))))
+
+(deftest corrupted-cards-will-fix-itself
+  (testing "#15222"
+    (testing "if card has an empty dataset_query, fixit up with an dummy query"
+      (mt/with-temp* [Card [{id1 :id} {:dataset_query ""}]
+                      Card [{id2 :id} {:dataset_query "{}"}]]
+        (is (= {:database (mt/id),
+                :native   {:query ""},
+                :type     :native}
+               (:dataset_query (db/select-one Card :id id1))))
```

Comment:
nit, you could `db/select-one-field` to save calling `:dataset_query` separately